src/Histogram.hpp

```diff
@@ -21,35 +21,59 @@ class Histogram: private std::vector<T> {
     using super::end;
 
     /// Default constructor
-    Histogram(): Histogram(0, 0, 0) {}
-    /// Constructor with a specific number of bins `nbins`, and which can hold
-    /// data in the `min - max` range.
-    Histogram(size_t nbins, double min, double max): super(nbins), min_(min), dr_((max - min) / nbins) {
+    Histogram(): Histogram(0, 0, 0, 0, 0, 0) {}
+    /// Constructor for a flat 2d histogram with a specific number of bins in each direction 
+    /// `n_x` and `n_y`, and which can hold data in the `min_x - max_x` range (resp `min_y - max_y`).
+    Histogram(size_t n_x, double min_x, double max_x, size_t n_y, double min_y, double max_y): super(n_x*n_y) {
+        nx_ = n_x;
+        min_x_ = min_x;
+        dx_ = (max_x - min_x) / n_x;
+        ny_ = n_y;
+        min_y_ = min_y;
+        dy_ = (max_y - min_y) / n_y;
         static_assert(
             std::is_arithmetic<T>::value,
             "Histogram<T> is only defined for arithmetics types T"
         );
     }
 
+    /// Constructor for a 1d histogram with a specific number of bins `n_bins`, and which can hold 
+    /// data in the `min - max` range.
+    Histogram(size_t n_bins, double min, double max): Histogram(n_bins, min, max, 1, 0, 1) {}
+
     Histogram(const Histogram&) = default;
     Histogram(Histogram&&) = default;
     Histogram& operator=(const Histogram&) = default;
     Histogram& operator=(Histogram&&) = default;
+    const T& operator()(size_t i, size_t j) const {
+        return (*this)[j + i * ny_];
+    }
 
     /// Get the size of the bins
-    double bin_size() const {return dr_;}
+    double dx() const {return dx_;}
```

Comment:
In order to be able to generate machine-readable doc with doxygen, can each function have its own documentation?